android/engine/src/main/java/org/smartregister/fhircore/engine/ui/pin/PinSetupActivity.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.ui.pin
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.WindowManager
+import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import org.smartregister.fhircore.engine.ui.appsetting.AppSettingActivity
+import org.smartregister.fhircore.engine.ui.base.BaseMultiLanguageActivity
+import org.smartregister.fhircore.engine.ui.login.LoginService
+import org.smartregister.fhircore.engine.ui.theme.AppTheme
+import org.smartregister.fhircore.engine.util.FORCE_LOGIN_VIA_USERNAME
+
+@AndroidEntryPoint
+class PinSetupActivity : BaseMultiLanguageActivity() {
+
+  @Inject lateinit var loginService: LoginService
+
+  val pinViewModel by viewModels<PinViewModel>()
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    loginService.loginActivity = this
+    window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE)
+
+    pinViewModel.apply {
+      loadData(isSetup = true)
+      val pinSetupActivity = this@PinSetupActivity
+      navigateToHome.observe(pinSetupActivity) { pinSetupActivity.moveToHome() }
+      navigateToSettings.observe(pinSetupActivity) { pinSetupActivity.moveToSettings() }
+      pin.observe(pinSetupActivity) { it.let { enableSetPin.postValue(it.length > 3) } }
```

Comment:
use >= and same constant or config value as https://github.com/opensrp/fhircore/pull/1030/files#diff-d63f836c3cd32ded90b5d79304289b00d7db17d9b2bb9d06f0a8e1f62156dc0eR73